vowpalwabbit/learner.h

```diff
@@ -194,16 +472,27 @@ struct learner
     else
     {
       increment_offset(ec, increment, lo);
-      learn_fd.multipredict_f(learn_fd.data, *learn_fd.base, (void*)&ec, count, increment, pred, finalize_predictions);
+      learn_fd.multipredict_f(
+          learn_fd.data, *learn_fd.base, static_cast<void*>(&ec), count, increment, pred, finalize_predictions);
       decrement_offset(ec, increment, lo);
     }
   }
 
+  template <class L, typename LabelT>
+  inline void set_predict(void (*u)(T&, L&, E&, LabelT&))
+  {
+    learn_fd.predict_with_label_f = (learn_data::fn_with_label)u;
```

Comment:
Is there some way to cross-check the label type with the typed-predict/learn functions?

Reply:
I've been thinking about that. I'm not sure of a good way we can guard this yet, but I imagine there's something we can do